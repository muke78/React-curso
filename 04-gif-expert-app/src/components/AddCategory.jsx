import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState("");

    const onInputChangue = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim())
        setInputValue("");
    };

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifts"
                value={inputValue}
                onChange={onInputChangue}
            />
        </form>
    );
};




AddCategory.propTypes = {
    onNewCategory: PropTypes.func
}

