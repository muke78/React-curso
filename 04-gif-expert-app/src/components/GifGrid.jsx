import PropTypes from "prop-types";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

/**
 * El useEffect se ocupa para disparar efectos secundario 
 * algun proceso que se quiera ejecutar cuando esto suceda
 */

/**
 * El useEffect cuando se le deja la lista de depencias
 * o el arreglo vacio significa que la primera vez
 * que cargue o se construta el componente solo lo hara una vez
 */

/**
 * El useEffect no se le pueden poner respuestas asincronas
 */


/**
 * Al pasar las props de un componente padre a uno hijo
 * se puede pasar por medio del operador spread para poder 
 * mandarle una copia exacta a todo lo que se renderice en
 * el hijo
 */