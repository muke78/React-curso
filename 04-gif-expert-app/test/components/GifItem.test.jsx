import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem/>', () => {

    const props = {
        url: "https://www.google.es/",
        title: "Link de google"
    }

    test('Debe de hacer la prueba del snapshot', () => {
        const { container } = render(
            <GifItem {...props} />
        );
        expect(container).toMatchSnapshot();
    })
    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(
            <GifItem {...props} />
        );
        // expect(screen.getByRole("img").src).toBe(props.url);
        const { src, alt } = screen.getByRole("img");
        expect(src).toBe(props.url);
        expect(alt).toBe(props.title);
    })
    test('debe de mostrar el titulo en el componente', () => {
        render(
            <GifItem {...props} />
        );
        expect(screen.getByText(props.title)).toBeTruthy();
    })
})