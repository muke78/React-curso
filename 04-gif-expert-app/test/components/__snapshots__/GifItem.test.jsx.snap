// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer la prueba del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Link de google"
      src="https://www.google.es/"
    />
    <p>
       
      Link de google
       
    </p>
  </div>
</div>
`;
